checker/jdk/signedness/src/java/lang/Integer.java

```diff
@@ -0,0 +1,1602 @@
+     *
+     * @param i the value whose bits are to be counted
+     * @return the number of one-bits in the two's complement binary
+     *     representation of the specified {@code int} value.
+     * @since 1.5
+     */
+    public static @NonNegative int bitCount(@UnknownSignedness int i) {
+        // HD, Figure 5-2
+        i = i - ((i >>> 1) & 0x55555555);
+        i = (i & 0x33333333) + ((i >>> 2) & 0x33333333);
+        i = (i + (i >>> 4)) & 0x0f0f0f0f;
+        i = i + (i >>> 8);
+        i = i + (i >>> 16);
+        return i & 0x3f;
+    }
+
+    /**
+     * Returns the value obtained by rotating the two's complement binary
+     * representation of the specified {@code int} value left by the
+     * specified number of bits.  (Bits shifted out of the left hand, or
+     * high-order, side reenter on the right, or low-order.)
+     *
+     * <p>Note that left rotation with a negative distance is equivalent to
+     * right rotation: {@code rotateLeft(val, -distance) == rotateRight(val,
+     * distance)}.  Note also that rotation by any multiple of 32 is a
+     * no-op, so all but the last five bits of the rotation distance can be
+     * ignored, even if the distance is negative: {@code rotateLeft(val,
+     * distance) == rotateLeft(val, distance & 0x1F)}.
+     *
+     * @param i the value whose bits are to be rotated left
+     * @param distance the number of bit positions to rotate left
+     * @return the value obtained by rotating the two's complement binary
+     *     representation of the specified {@code int} value left by the
+     *     specified number of bits.
+     * @since 1.5
+     */
+    public static @PolySigned int rotateLeft(@PolySigned int i, int distance) {
+        return (i << distance) | (i >>> -distance);
+    }
+
+    /**
+     * Returns the value obtained by rotating the two's complement binary
+     * representation of the specified {@code int} value right by the
+     * specified number of bits.  (Bits shifted out of the right hand, or
+     * low-order, side reenter on the left, or high-order.)
+     *
+     * <p>Note that right rotation with a negative distance is equivalent to
+     * left rotation: {@code rotateRight(val, -distance) == rotateLeft(val,
+     * distance)}.  Note also that rotation by any multiple of 32 is a
+     * no-op, so all but the last five bits of the rotation distance can be
+     * ignored, even if the distance is negative: {@code rotateRight(val,
+     * distance) == rotateRight(val, distance & 0x1F)}.
+     *
+     * @param i the value whose bits are to be rotated right
+     * @param distance the number of bit positions to rotate right
+     * @return the value obtained by rotating the two's complement binary
+     *     representation of the specified {@code int} value right by the
+     *     specified number of bits.
+     * @since 1.5
+     */
+    public static @Unsigned int rotateRight(@Unsigned int i, int distance) {
```

Comment:
Why is rotateRight unsigned->unsigned whereas rotateLeft is polymorphic?  I think both should be polymorphic.

Reply:
That might look suspicious.
Check [this](https://github.com/typetools/checker-framework/pull/2523#discussion_r291809482).

Reply:
Your response is not relevant to the comment, which is about the discrepancy between the annotations on rotateLeft and rotateRight.  I believe that they should be annotated similarly.  Do you agree with that or disagree with it?
Your response might be relevant to deciding how to annotate both rotateLeft and rotateRight.  What annotation do *you* believe is correct for the methods, and why?

Reply:
>  I believe that they should be annotated similarly. Do you agree with that or disagree with it?

I agree
> Your response might be relevant to deciding how to annotate both rotateLeft and rotateRight. What annotation do _you_ believe is correct for the methods, and why?

I believe that both should be annotated with `@PolySigned` as both rotate the bits which can be unsigned or signed. There was an issue related to this, therefore, I was confused. I will correct it.